src/main/js/isd.js

```diff
@@ -59,23 +59,59 @@
 
         };
 
-        /* process regions */
+        /* Filter body contents - Only process what we need within the offset and discard regions not applicable to the content */
+        var body = {};
+        var activeRegions = new Set();
+
+        function filter(offset, element) {
+            function offsetFilter(element) {
+                return !(offset < element.begin || offset >= element.end);    
+            }    
+        
+            if (element.contents) {
+                var clone = {};
+                for (var prop in element) {
+                    clone[prop] = element[prop];
+                }
+                clone.contents = [];
 
-        for (var r in tt.head.layout.regions) {
+                element.contents.filter(offsetFilter).forEach(function (el) {
+                    var filteredElement = filter(offset, el);
+                    if (filteredElement.regionID) {
+                        activeRegions.add(filteredElement.regionID);
+                    }
+        
+                    if (filteredElement !== null) {
+                        clone.contents.push(filteredElement);
+                    }
+                });
+                return clone;
+            } else {
+                return element;
+            }
+        }
 
+        body = filter(offset, tt.body);
+
+        /* rewritten TTML will always have a default - this covers it. because the region is defaulted to "" */
+        if (activeRegions.size === 0 && tt.head.layout.regions.hasOwnProperty("")) {
+            activeRegions.add("");
+        }
+        
+        /* process regions */      
+
+        activeRegions.forEach(function (regionID) {
```

Comment:
suggest using `for(..;..;..)` for consistency with the rest of the codebase.

Reply:
I don't understand how this could work?

Unlike many places where `for(..;..;..)` is used, `activeRegions` is a `Set`, so changing to a `for()` construct would just increase verbosity without making the code any easier to understand. We'd have to construct an iterator, and call `next()` on it, testing for it it being non-empty and referencing `iterator.value` every time we want to use it. Seems like a significant drop-off for the sake of something sort-of-approaching consistency.

Happy to reconsider if you think it's worth it @palemieux .

Reply:
Ah. imscJS was trying to stick with ES5, and so no `Set()`.

Two options:

* replace with objects `{}`
* move to ES6

Reply:
Strange, the `grunt build` process bundles back into pre-es6 and seems to deal with `Set()` fine. Guessing that it's converting it to another form.

Reply:
Oh. Let me check that...

Reply:
Have replaced with `{}` in f637d5a.